### PR TITLE
using lld linker

### PR DESCRIPTION
**Pull request by Guo-Peilin**

### CODE REVIEW

**Comment by qiuxiafei** [comment]
Do we need to install lld in docker ?

File: tao/.bazelrc [hunk 1]
```diff
@@ -37,7 +37,7 @@ build:disc --config=release_base
 
 build:disc_cpu --config=disc --cxxopt=-DTAO_CPU_ONLY
 build:disc_x86 --config=disc_cpu --config=release_cpu_linux --cxxopt=-DTAO_X86 --define disc_x86=true
-build:disc_aarch64 --config=disc_cpu --cxxopt=-DTAO_AARCH64 --define disc_aarch64=true --linkopt="-Xlinker --stub-group-size -Xlinker 10000000"
```
Comment (wyzero) [inline tao/.bazelrc:40]: Why remove these flags?

Comment (Guo-Peilin) [reply tao/.bazelrc:40]: lld do not recognize this flag

**Comment by Guo-Peilin** [comment]
>

**Comment by Guo-Peilin** [comment]
> Do we need to install lld in docker ?

will use default `ld.gold` if `lld` not installed

File: scripts/python/tao_build.py [hunk 2]
```diff
@@ -333,6 +334,10 @@ def bazel_build(target, flag=""):
         if args.skip_compute_intensive_fusion:
             flag += ' --config=skip_compute_intensive_fusion'
 
+        if hasLLD():
```
Comment (qiuxiafei) [inline scripts/python/tao_build.py:337]: could all `if hasLLD()`s in each file be in a common place?

**Comment by qiuxiafei** [comment]
Do we need to install `lld` in our dev docker image by default?

File: pytorch_blade/bazel_build.py [hunk 3]
```diff
@@ -152,6 +152,8 @@ def __init__(self, *args, **kwargs):
         if running_on_ci():
             self.configs += ["--config=ci_build"]
 
+        self.configs += [" --linkopt='-fuse-ld=gold -Xlinker --stub-group-size -Xlinker 10000000'"]
```
Comment (qiuxiafei) [inline pytorch_blade/bazel_build.py:155]: Nit: We could extend `extra_link_flags` to receive an default argument like `force_gold=False` or whatever, to encapsulate gold default argument into a single place.

**Review by qiuxiafei — APPROVED** [review]
LGTM with minor comments~

**Review by qiuxiafei — APPROVED** [review]

**Pull request merged**